DataProcessing/dataHandler.py

```diff
@@ -0,0 +1,125 @@
+import os.path
+import os
+import os.path as osp
+import tempfile
+from argparse import ArgumentParser
+
+import mmcv
+import numpy as np
+from mmtrack.apis import inference_mot, init_model
+from cv2 import imwrite
+import os, glob, shutil
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from sklearn.cluster import KMeans
+from sklearn.metrics import silhouette_score
+import cv2
+
+class DataHandler:
+
+    def __init__(self, 
+                 input_folder_path:str,
+                 output_video_path:str,
+                 config,
+                 checkpoint,
+                 device,
+                 ):
+
+        self.input_path = input_folder_path
+        self.output_path = output_video_path
+        self.config = config
+        self.checkpoint = checkpoint
+        self.device = device
+        self.crop_folder_path = self.output_path + 'crops/'
+        self.cluster_folder_path = self.output_path + 'clusters/'
+        self.cam_id = 2
+
+    def track_persons_and_crop_from_dir(self, capture_index:int=500, acc_threshold:int=0.999) -> None:
+        """
+        capture_index = frame will be captured every capture_index value. input 1 for all frames
+        """
+        def create_crops_folder() -> None:
+            """Creates a crops folder for the given output path"""
+            assert self.output_path
+            if not osp.isdir(self.crop_folder_path):
+                os.mkdir(self.crop_folder_path)
+
+        create_crops_folder()
+        if osp.isdir(self.input_path):
+            vids = os.listdir(self.input_path)
+            print(f'Found the following videos to process: {vids}')
+            for vid in vids:
+                cur_path = os.path.join(self.input_path, vid)
+                print(f'Processing video: {cur_path}')
+                if not osp.isdir(cur_path) and '.avi' in cur_path:
+                    imgs = mmcv.VideoReader(cur_path)
+                    print('begin extraction of video {}'.format(vid))
+                    model = init_model(self.config, self.checkpoint, device=self.device)
+                    prog_bar = mmcv.ProgressBar(len(imgs))
+                    # test and show/save the images
+                    for i, img in enumerate(imgs):
+                        if isinstance(img, str):
+                            img = osp.join(self.input_path, img)
+                        prog_bar.update()
+                        cap = 0
+                        if i % capture_index == 0:
+                            cap = 10
+                        if cap > 0:  # we capture 10 frames every 500 frames
```

Comment:
If I'm not mistaken this didn't work last time I checked, we take only one frame and not 10

Reply:
True, cant take modulo of 0. can be fixed but I figure we can drop this idea.